streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -33,14 +28,11 @@
 
     final void transitionTo(final Task.State newState) {
         final State oldState = state();
-        if (oldState == CREATED && (newState == RESTORING || newState == CLOSING)) {
-        } else if (oldState == RESTORING && (newState == RUNNING || newState == SUSPENDED || newState == CLOSING)) {
-        } else if (oldState == RUNNING && (newState == SUSPENDED || newState == CLOSING)) {
-        } else if (oldState == SUSPENDED && (newState == RUNNING || newState == CLOSING)) {
-        } else if (oldState == CLOSING && (newState == CLOSING || newState == CLOSED)) {
+
+        if (oldState.isValidTransition(newState)) {
```

Comment:
Further simplified this transition -- suspended must first transit to restoring and then running.